Webui: prepared 4 master

zeners opened the pull request:
alle Funktionen für Master implementiert.

Irgendwie spinnt das Erstellen von pgp-linked Identitäten. Evtl. kannst Dir das ja mal bitte anschauen.

csoler committed:
fixed compilation of circles

csoler committed:
updated branch to upstream/master

csoler committed:
added circle edit functionality to People tab

csoler committed:
move circles code to IdDialog

csoler committed:
fixed some gui bits in IdDialog

csoler committed:
incorporated defnax changes

defnax committed:
Improved layout better look & feel for Circles View
Added activity column

defnax committed:
Clean up naming and added to display avatar icons.

defnax committed:
Added Context menus for Edit Circle or Add/Remove Member
Changed for OK Button renaming to Create (New Mode) or Update(Edit Mode)

csoler committed:
changed serialisation methods to use c++ in rsnxsitems, added two new items for encrypted data, renamed classes RsNxsSyncGrp and RsNxsSyncMsg to RsNxsSyncGrp/MsgReqItem

csoler committed:
fixed bug in deserialisation causing the wrong method to be called

csoler committed:
Merge pull request #3 from defnax/v0.6-Circles

Improved layout better look & feel for Circles View

csoler committed:
added serialisation for encrypted items

csoler committed:
started with new crypto methods for multiple peers

csoler committed:
half way through RsNxsItem encryption

csoler committed:
finished implementing decryption routines. Still needs fixing compilation

csoler committed:
modifications to item encryption so as to match APIs

electron128 committed:
- make tests compile
- added tests to travis.yml
- excluded grouteritems and photoitems from test, because they changed too much
- disabled failing tests
- all tests pass in valgrind, without valgrind result is undefined
- to compile tests add CONFIG+=tests to qmake args. Then run tests/unittests/unittests

electron128 committed:
added CONFIG+=NO_SQLCIPHER support to unittests (travis wants this)

electron128 committed:
redirect travis testing output to /dev/null, because the logfile was to long

electron128 committed:
fixed uninitialised memory read in test libretroshare_gxs.gxs_grp_sync_delayed

electron128 committed:
added missing lock in RsSharedPtr destructor

electron128 committed:
added missing mutex in nxstesthub

electron128 committed:
enabled RsNxsItem test

electron128 committed:
attempt to fix output redirect for unittests in travis

electron128 committed:
added missing initialisation of local, not serialised msg meta data in NxsTestHub

electron128 committed:
fixed threading issue in NxsTestHub. NxsTesthub has to wait for its own thread to shut down on test end.

electron128 committed:
exclude not serialised member from operator== for RsNxsTransac

electron128 committed:
fix compile for new gxs metadata serialisation

electron128 committed:
give gxs grp sync test more time, because of sync and timestamp update cycle times

electron128 committed:
disabled tlvrandom test because it is buggy and fails sometimes

csoler committed:
merged with latest trunk

csoler committed:
fixed compilation of RsNxsTransaction encryption code

csoler committed:
updated v0.6-Circles with latest trunk

csoler committed:
added debug output for encryption/decryption of transactions

csoler committed:
updated to upstream/master

csoler committed:
merged with upstream/master

csoler committed:
fixed merge

csoler committed:
fixed merge

csoler committed:
merged with upstream

csoler committed:
fixed ui file for IdDialog

csoler committed:
fixed compilation

csoler committed:
called re-loading of circles after creation. Added debug info in msg vetting

defnax committed:
increase default font size for the ListWidget.

electron128 committed:
disabled gxs_grp_sync test, because it fails after rebase to master

electron128 committed:
remove wrong info about webui from readme (webui settings in gui does not have and effect on rs-nogui)

electron128 committed:
added info about tests to readme

csoler committed:
fixed sending part in encrypted msg items

csoler committed:
debugged transport of encrypted items. Still needs work.

csoler committed:
added single-block multiple encryption

PhenomRetroShare committed:
Fix error "(EE) SSGxsChannelGroup::load() asked to load a null string.
Weird." by creating a default serviceString if empty.

csoler committed:
added code for single item encryption/decryption

csoler committed:
fixed encryption method on the basis that encrypted keys always have the same length

csoler committed:
added proper decryption calls for partial decryption and possible re-try after loading keys.

csoler committed:
fixed one bug in debug output of encrypted key offset

csoler committed:
implemented per-item encryption and Vetting method compatibility layer

csoler committed:
fixed small bug causing update of TS to fail in encrypted NxsItems

csoler committed:
removed dead code in GxsSecurity

zeners committed:
webui: chat: unread msg count for peer fixed

zeners committed:
webui: lobby-chat subscribe fixed

electron128 committed:
Merge pull request #202 from electron128/fixtests

make tests compile

electron128 committed:
fix format in readme

csoler committed:
moved subscription flags when received group changes/create group to the case where the group is actually created (fixing bug causing group meta data change to unsubscribe friends)

csoler committed:
improved debug output in gxsnetservice.cc

csoler committed:
removed debug output from p3grouter.cc

csoler committed:
removed compilation warning and added warning about missing vetting for external circles

csoler committed:
added missing -lpthread

Commit:
impoved cleanup of unused locations

PhenomRetroShare committed:
Define empty service string as a valid case.

Commit:
removed unused map lookups

csoler committed:
Merge pull request #290 from sehraf/pr-improved-unused-location-cleanup

impoved cleanup of unused locations

csoler committed:
added update of mServerUpdateTS up to last mRecvTS of each group, which holds the last time the grp meta data was received/modified

csoler committed:
added copy of previous subscription flags when group update is received. Avoids unsubscribing friends when publishing an update to grp metadata

csoler committed:
added notification for modified metadata in group. Does not yet work on friends though.

defnax committed:
Merge branch 'master' of https://github.com/RetroShare/RetroShare

defnax committed:
update todo

csoler committed:
added missing update of mRecvTS in database when updating a group metadata

csoler committed:
merged with upstream/master

electron128 committed:
Merge pull request #285 from PhenomRetroShare/Fix_SSGxsChannelGroupLoadNullError

Fix error "(EE) SSGxsChannelGroup::load() asked to load a null string.

csoler committed:
attempt at fixing negative bw rates in GUI that may pop up when an error is present in the SSL stream (e.g. when a peer disconnects/reconnects)

csoler committed:
added test against empty circles

csoler committed:
added missing collection of some peers in p3gxscircles::recipients()

csoler committed:
updated ubuntu changelog

csoler committed:
fixed variable scope problem in introserver.cc

csoler committed:
added edit of existing circles

csoler committed:
improved circles GUI logic

csoler committed:
fixed removal of peers from circles

csoler committed:
added TODO file specific to circles

csoler committed:
updated circles todo

csoler committed:
updated circles TODO

csoler committed:
disabled auto-sync for messages in GXS circles, while keeping Grp auto-sync

csoler committed:
updated TODO(circles), removed useless context menu, added tooltips for circle distribution options

csoler committed:
added ban menu entry to chat lobby participant list

csoler committed:
updated circles TODO

csoler committed:
fixed display of info for non admin circles, and added readonly system to not allow editing them

csoler committed:
updated Circles TODO

PhenomRetroShare committed:
Add a new column in RsColl Editor to get only name in first one.

PhenomRetroShare committed:
Fix Paste RetroShare Link action when copy File link.

csoler committed:
Merge pull request #302 from PhenomRetroShare/Fix_RsLinkCopyForExtraFile

Fix Paste RetroShare Link action when copy File link.

csoler committed:
Merge pull request #301 from PhenomRetroShare/Add_NewColumnInRsCollForPath

Add a new column in RsColl Editor to get only name in first one.

csoler committed:
allow non signature checking for GR message recepts at intermediate peers when key is missing. Should fix return receipt being lost in the way keeping messages in outbox

csoler committed:
fixing compilation problem

csoler committed:
fixed culumn name in GR stats

csoler committed:
fixed auto-subscribe of circles when unsigned identities are locally owned

PhenomRetroShare committed:
Correct Colored Name Contrast in chat, same way than text.

csoler committed:
Merge pull request #303 from PhenomRetroShare/Fix_CorrectColoredNameContrast

Correct Colored Name Contrast in chat, same way than text.

csoler committed:
merged

csoler committed:
improved one line of debug output in turtle

csoler committed:
added highlighting of currently selected circle members

zeners committed:
Merge branch 'master'

csoler committed:
unselect all IDs when no circle is selected

csoler committed:
cleaned code a little bit

zeners committed:
webui: route peer location to chat-lobby

zeners committed:
webui: need for master

zeners committed:
webui: unread chat message counter in menu

zeners committed:
webui: chat-layout fixed

zeners committed:
webui: unsubscribe chat-lobby

csoler committed:
fixed circle auto-subscribe

csoler committed:
changed titles of tree root nodes in circles

csoler committed:
improved debugDump(), showing the subscription status of each grp

electron128 commented:
Das Problem mit den pgp linked identities ist: zum signen braucht man den PGP key, und um da ran zu kommen braucht man das Passwort. libretroshare erlaubt im Moment nur eine Quelle für das Passwort. Bei rs-gui ist es die gui, bei rs-nogui die webui (Konsole geht inter über das webui backend).
Normal ist das auch kein Problem, weil das Passwort nur beim Start gebraucht wird. Und beim Start ist die Kontrolle sowieso exclusiv bei gui oder webui.

Man müsste das Handling des Passwortes asynchron machen, so dass auch andere Threads als der GUI Thread nach dem Passwort fragen können. Und dass mehrere Quellen das Passwort liefern können, je nachdem welche Quelle schneller ist.

zeners commented:
Sowas hatte ich befürchtet. PGP-Linked Identitäten für den Webclient vorerst deaktivieren?

csoler committed:
set default value for circle distribution type

csoler committed:
fixed effect of selecting known/signed IDs in circle creation dialog

csoler committed:
attempt at fixing circle cache loop. Renamed a few variables and fixed logic. Not yet fully working

csoler committed:
fixed potential crash due to bad usage of updates

csoler committed:
added test to only subscribe/unsubscribe circles when needed, hence avoiding an unnecessary TS update

csoler committed:
added force reload of cache for circles that are modified

dartraiden committed:
correct path to Hidden Service configuration panel

csoler committed:
Merge pull request #308 from dartraiden/master

Сorrect path to Hidden Service configuration panel

PhenomRetroShare committed:
Fix delete in PictureFlow

csoler committed:
Merge pull request #4 from PhenomRetroShare/Fix_PictureFlowDeletion

Fix delete in PictureFlow

csoler committed:
added update of GUI, instead of re-creating everything when a circle change

csoler committed:
added reload of grp meta on update

csoler committed:
fixed qt4 compilation

csoler committed:
fixed bug causing GUI to not auto-unsubscribe in circles

csoler committed:
fixed update of GUI when circles change

csoler committed:
fixed creation/distribution of self-restricted circles

csoler committed:
fixed update of circle name in the GUI

csoler committed:
fixed mixup between in/out bandwidth traffic info

zeners committed:
Merge remote branch 'origin/master'

zeners committed:
webui: pgp-linked identity disabled

reason: password-request don't work

PhenomRetroShare committed:
Add Plugins Path for Apple

From Jenster

csoler committed:
Merge pull request #312 from PhenomRetroShare/AddPluginsPathForApple

Add Plugins Path for Apple

csoler committed:
used RsTemporaryMemory to avoid memory leak in case of error in GxsTunnels

PhenomRetroShare committed:
Fix error reported by CppChecker:

/libretroshare/src/gxs/rsgenexchange.cc:2850: erreur : Cppcheck:
Dereferencing 'grp' after it is deallocated / released
/libretroshare/src/gxstunnel/p3gxstunnel.cc:1004: erreur : Cppcheck:
Invalid usage of output stream: '<< std::cerr'.
/libretroshare/src/util/contentvalue.cc:185: erreur : Cppcheck: Memory
leak: dest
/openpgpsdk/src/openpgpsdk/reader_armoured.c:575: erreur : Cppcheck:
Common realloc mistake: 'buf' nulled but not freed upon failure
/retroshare-gui/src/gui/FileTransfer/DLListDelegate.cpp:103: erreur :
Cppcheck: Array 'byteUnits[4]' accessed at index 4, which is out of
bounds.
/retroshare-gui/src/gui/FileTransfer/DLListDelegate.cpp:122: erreur :
Cppcheck: Array 'byteUnits[4]' accessed at index 4, which is out of
bounds.
/retroshare-gui/src/gui/FileTransfer/DLListDelegate.cpp:141: erreur :
Cppcheck: Array 'byteUnits[4]' accessed at index 4, which is out of
bounds.
/retroshare-gui/src/gui/FileTransfer/ULListDelegate.cpp:92: erreur :
Cppcheck: Array 'byteUnits[4]' accessed at index 4, which is out of
bounds.
/retroshare-gui/src/gui/FileTransfer/ULListDelegate.cpp:111: erreur :
Cppcheck: Array 'byteUnits[4]' accessed at index 4, which is out of
bounds.
/retroshare-gui/src/gui/People/PeopleDialog.cpp:774: erreur : Cppcheck:
Uninitialized variable: layout
/retroshare-gui/src/gui/People/PeopleDialog.cpp:864: erreur : Cppcheck:
Uninitialized variable: layout
/retroshare-gui/src/gui/common/AvatarWidget.cpp:225: erreur : Cppcheck:
Uninitialized variable: status
/retroshare-nogui/src/rpc/proto/rpcprotostream.cc:805: erreur :
Cppcheck: Memory leak: buffer
/supportlibs/pegmarkdown/GLibFacade.c:150: erreur : Cppcheck: va_list
'args' was opened but not closed by va_end().
/openpgpsdk/src/openpgpsdk/reader_armoured.c:575: erreur : Cppcheck:
Common realloc mistake: 'buf' nulled but not freed upon failure
/tests/unittests/libretroshare/serialiser/rsturtleitem_test.cc:213:
erreur : Cppcheck: Return value of allocation function 'init_item' is
not stored.

csoler committed:
Merge pull request #314 from PhenomRetroShare/Fix_CppCheckerError

Fix error reported by CppChecker:

dartraiden committed:
Fix typo

csoler committed:
Merge pull request #315 from dartraiden/master

Fix typo

csoler committed:
fixed display of properties in restricted groups

csoler committed:
updated TODO

realityfabric committed:
Delete empty BUGS file

zeners committed:
Merge branch 'master'

zeners committed:
webui: options / rights (only defaults)

zeners commented:
hab mal an den Optionen herum geschraubt. Da es bereits Lese-Zugriff auf die Berechtigungen gab, hab ich ein wenig im server-api herumgespielt: bitte meine Änderungen da mal gegenprüfen, ich bin alles andere als ein c++ Entwickler. Aber es tut. :)

csoler committed:
update of groups up to the latest attached circle server update TS. Should re-send groups that depend on a circle when the circle change

zeners committed:
webui: updated mithril to v0.2.3

no idea what it was before

zeners committed:
webui: view / change rights per user

zeners committed:
webui: chat: display peer location

zeners committed:
undo gitignore

zeners committed:
webui: RS-reboot / RS-reboot-detection improved

paths without token, are now pulled

zeners commented:
hab im retroshare.js (rs) herum geschraubt: paths, die im cache liegen, aber kein Token haben, werden jetzt gepullt

zeners committed:
webui: options>settings, playing with colors

csoler committed:
fixed potential crash due to using deleted data (found by dimqua)

zeners committed:
webui: integrate new webui in unix build process

zeners commented:
qmake Integration mal soweit fertig, dass sie funktioniert.
allerdings spinnt qmake ziemlich rum, wenn es dateien installieren soll, die es beim lauf von qmake noch nicht gibt. Daher werden die Files AUCH beim qmake bereits erzeugt (ein touch würde reichen, aber dazu brauche ich dann eine extra-routine)

eventuell können wir die generierten files auch direkt in der Ausgabe erzeugen lassen und nicht zwingend zwischen die Sourcen schreiben (beim Kompilieren halt, für die Entwicklung lassen wir es, wie es ist)

csoler committed:
added anon ids to offline vetting

csoler committed:
removed deadlock

csoler committed:
added disabling of combo elements in GxsIdChooser for creating forums posts with unautorised GXS ids. Still needs a bit of work

zeners committed:
webui: switch from react to mithril

csoler committed:
fixed async-ed circle loading in IdDialog

csoler committed:
fixed display of warning in posting into forums without proper credentials

csoler committed:
cleaned debug info and disabled local circles before merge

csoler committed:
fixed small bug in debug output; cleanup debug output

csoler committed:
small code cleaning bits

csoler committed:
removed some debug info

csoler committed:
improved display of circle creator ID

csoler committed:
fixed tiny bits in circles/people GUI

csoler committed:
merged with upstream/master

csoler committed:
turned PeopleDialog to unix ff

zeners committed:
webui: paste RS-links for download

csoler committed:
fixed unit tests

csoler committed:
Merge pull request #325 from csoler/v0.6-Circles

V0.6 circles

zeners committed:
Merge branch 'master'

zeners committed:
webui: add downloads: focus fixed

csoler committed:
moved discarding of banned identities from lobbies before signature checking, so as to avoid to time stamp their GXSId keys

zeners committed:
Merge branch 'master'

zeners committed:
webui: windows creating batch-files

zeners committed:
webui: searchresult cross-checking with downloads

zeners committed:
webui: removed searchresult debug output

electron128 committed:
Merge pull request #323 from zeners/webui

New webui frontend based on Mithril.js. Thx to Zener.

zeners committed:
webui 4 windows build prepare

should be testet by someone

zeners committed:
Merge branch 'master'

zeners committed:
webui: menu in gui mode fixed

electron128 committed:
Merge pull request #327 from zeners/webui

Webui

PhenomRetroShare committed:
Fix WebUI Make Source build script for LMDE and other using /bin/sh.

csoler committed:
updated ubuntu changelog

csoler committed:
added xenial as build target

thunder2 committed:
Fixed Windows compile.

zeners committed:
Merge branch 'master'

zeners committed:
webui: include files in install

electron128 committed:
Merge pull request #330 from zeners/webui

fixed missing webui files in make install

electron128 committed:
Merge pull request #328 from PhenomRetroShare/Fix_WebUI_MakeSrcBuild

Fix WebUI Make Source build script for LMDE and other using /bin/sh.

electron128 committed:
Merge pull request #318 from realityfabric/remove_empty_file

Delete empty BUGS file

zeners committed:
compatible to g++11 / QT5

method suggested by cyril

csoler committed:
Merge pull request #332 from zeners/master

compatible to g++11 / QT5

defnax committed:
Fixed the default splitter expanding of People Dialog

zeners committed:
webui: patch init.sh like build.sh

csoler committed:
Merge pull request #333 from zeners/webui

webui: patch init.sh like build.sh

zeners committed:
webui: long usernames display fixed
